README.md

```diff
@@ -0,0 +1,335 @@
+
+```go
+type Recipe struct {
+	Id          primitive.ObjectID `bson:"_id"`
+	Name        string             `bson:"name"`
+	Ingredients []string           `bson:"ingredients"`
+}
+```
+
+With the MongoDB Go driver, we can annotate Go data structures with BSON so that way we can easily map between the two. It essentially makes our lives a lot easier when working with MongoDB and Go.
+
+Let's circle back to the `IntentDispatcher` function:
+
+```go
+func (connection Connection) IntentDispatcher(ctx context.Context, request alexa.Request) (alexa.Response, error) {
+	var response alexa.Response
+	switch request.Body.Intent.Name {
+	case "GetIngredientsForRecipeIntent":
+	case "GetRecipeFromIngredientsIntent":
+	default:
+		response = alexa.NewSimpleResponse("Unknown Request", "The intent was unrecognized")
+	}
+	return response, nil
+}
+```
+
+Remember the two intents from the Alexa Developer Portal? We need to assign logic to them.
+
+Essentially, we're going to do some database logic and then use the `NewSimpleResponse` function to create a response the the results.
+
+Let's start with the `GetIngredientsForRecipeIntent` logic:
+
+```go
+case "GetIngredientsForRecipeIntent":
+    var recipe Recipe
+    recipesCollection := connection.database.Collection("recipes")
+    recipeName := request.Body.Intent.Slots["recipe"].Value
+    if recipeName == "" {
+        return alexa.Response{}, errors.New("Recipe name is not present in the request")
+    }
+    if err := recipesCollection.FindOne(ctx, bson.M{"name": recipeName}).Decode(&recipe); err != nil {
+        return alexa.Response{}, err
+    }
+    response = alexa.NewSimpleResponse("Ingredients", strings.Join(recipe.Ingredients, ", "))
+```
+
+In the above snippet, we are getting the slot variable that was passed and are issuing a `FindOne` query against the collection. The filter for the query says that the `name` field of the document must match the recipe that was passed in as a slot variable.
+
+If there was a match, we are serializing the array of ingredients into a string and are returning it back to Alexa. In theory, Alexa should then read back the comma separated list of ingredients.
+
+Now let's take a look at the `GetRecipeFromIngredientsIntent` intent logic:
+
+```go
+case "GetRecipeFromIngredientsIntent":
+    var recipes []Recipe
+    recipesCollection := connection.database.Collection("recipes")
+    ingredient1 := request.Body.Intent.Slots["ingredientone"].Value
+    ingredient2 := request.Body.Intent.Slots["ingredienttwo"].Value
+    cursor, err := recipesCollection.Find(ctx, bson.M{"ingredients": bson.D{{"$all", bson.A{ingredient1, ingredient2}}}})
+    if err != nil {
+        return alexa.Response{}, err
+    }
+    if err = cursor.All(ctx, &recipes); err != nil {
+        return alexa.Response{}, err
+    }
+    recipeList := ""
+    for _, recipe := range recipes {
+        recipeList += recipe.Name
+    }
+    response = alexa.NewSimpleResponse("Recipes", recipeList)
```

Comment:
`recipeList` will be one string with no delimiters but the `GetIngredientsForRecipeIntent` intent returns one comma-delimited string created via `strings.Join`. Are both correct? Even if they are both correct, maybe we should stick to one method for consistency?